planner/core/find_best_task.go

```diff
@@ -279,13 +279,48 @@ func (p *baseLogicalPlan) enumeratePhysicalPlans4Task(physicalPlans []PhysicalPl
 		}
 		opt.appendCandidate(p, curTask.plan(), prop)
 		// Get the most efficient one.
-		if curTask.cost() < bestTask.cost() || (bestTask.invalid() && !curTask.invalid()) {
+		if curIsBetter, err := compareTask(p.ctx, curTask, bestTask); err != nil {
+			return nil, 0, err
+		} else if curIsBetter {
 			bestTask = curTask
 		}
 	}
 	return bestTask, cntPlan, nil
 }
 
+func compareTask(ctx sessionctx.Context, curTask, bestTask task) (curIsBetter bool, err error) {
```

Comment:
Add some comments about the return value of the function.

Reply:
Some comments are added, PTAL